La formazione per autodidatti

```diff
@@ -0,0 +1,77 @@
+Guardare il monitor e vedere schermate di codice che scorrono sotto i vostri occhi non è sufficiente; è necessario che quel codice lo vediate con “gli occhi della mente” e che nella vostra testa si delinei subito il quadro complessivo in cui quel codice si innesta.
+Meglio ancora: dovreste poter guardare il codice –sempre con gli occhi della mente-, e dovreste essere in grado di immaginare scenari diversi, dovreste sentire una voce interna che vi dice “ah, si, questo codice fa questa cosa ma si può fare meglio/più rapidamente/con meno dispendio di energie e risorse se facessi in quest’altro modo…”
+In due parole: se non siete “naturalmente portati” verso la materia, potete farcela lo stesso, ma la vostra vita professionale sarà una fatica perenne.
+Vi conviene? Solo perché avete in mente un’immagine “romantica” del programmatore e del suo mestiere, magari tratta dai numerosi film che potreste aver visto?
+Non che il mestiere in se’ sia gelido, tutt’altro: a volte l’eleganza di una soluzione ad un problema, può commuovere fino alle lacrime e non per densità semantica condensata in due righe di codice ma per l’idea che dietro quelle due righe di codice c’è stato un guizzo creativo, subito accompagnato da finezza nella tecnica, ma pur sempre prodotto da un essere umano: questo, è il bello ed il romantico di questo mestiere.
+Inutile dire che di sola romanticheria non si vive e che spesso e volentieri, anche utilizzando gli strumenti più appropriati e più adatti per sollevarci da lavori ripetitivi, la vita quotidiana di un programmatore non è sistematicamente soggetta a scariche di adrenalina.
+Anzi, dietro un lavoro ben fatto, di norma, c’è tanta, tanta “falegnameria”.
+Insomma, se alle domande “ho una reale predisposizione?” e “sono disposto ad accettare che il percorso di istruzione che sto per intraprendere non sarà tutto un tappeto di petali di rose?” avete risposto con un SI netto e sicuro, siete già a metà dell’opera, congratulazioni!
+
+Tutti possiamo programmare, allora?
+In teoria, nel momento stesso in cui pensate “ah, ora vado a mettere su la caffettiera”, voi tutti avete in mente –senza rendervene conto-, un piano precisissimo che consiste nei passi di 
+
+1)	Prendere la caffettiera
+2)	Riempire d’acqua la caldaia fino al segno
+3)	Inserire il filtro
+4)	Svuotare dell’eccesso di acqua
+5)	Riempire il filtro con il caffè senza pressarlo
+6)	Ecc…. fino alla tazzina fumante
+
+Non tutti i problemi si risolvono altrettanto facilmente ma spesso, più che la competenza tecnica, quello che fa un’enorme differenza è l’approccio al problema e –di riflesso-, anche alla sua soluzione.
+Esempio banale ma calzante: una volta mi è capitato di avere a che fare con un’applicazione discretamente complessa nella quale, ad un certo punto del processo, avrei dovuto stabilire in un loop molto pesante se nel database dovevo effettuare un’operazione di INSERT o di UPDATE.
```

Comment:
```suggestion
Esempio banale ma calzante: una volta mi è capitato di avere a che fare con un’applicazione discretamente complessa nella quale, ad un certo punto del processo, avrei dovuto stabilire in un loop molto pesante se nel database dovevo effettuare un’operazione di inserimento o di aggiornamento dei dati.
```

In relazione a quanto già detto da @corradopetrelli, propongo la mia suggestion (ma sentiti libero di scartarla qualora avessi delle proposte migliori)